src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -42,8 +49,51 @@ private static void initHttpClient()
         null,
         new DefaultHostnameVerifier());
 
+
+    /**
+     * Use a anonymous class to implement
+     * the interface ServiceUnavailableRetryStrategy()
+     * The max retry time is 3.
+     * The interval time is backoff.
+     */
     httpClient = HttpClients.custom()
         .setSSLSocketFactory(f)
+        .setServiceUnavailableRetryStrategy(new ServiceUnavailableRetryStrategy()
+        {
+          private int executionCount = 0;
+          @Override
+          public boolean retryRequest(
+              final HttpResponse response, final int executionCount,
+              final HttpContext context)
+          {
+            this.executionCount = executionCount;
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            boolean needNextRetry = (statusCode == 408 || statusCode >= 500 )
+                && executionCount < 4;
+            if(executionCount == 4)
```

Comment:
Instead of numeric constants in code, I'd recommend using variables above. So this should read something like 
boolean needNextRetry = ... & executionCount < MAX_RETRIES + 1, and define MAX_RETRIES = 3 above.
Do same for status code 408.